Source/Core/DolphinQt/GameList/GameListModel.cpp

```diff
@@ -57,6 +58,7 @@ QVariant GameListModel::data(const QModelIndex& index, int role) const
     return QVariant();
 
   const UICommon::GameFile& game = *m_games[index.row()];
+  TimePlayed timer;
```

Comment:
This line looks very innocuous but actually does disk IO and parses an ini file. Probably not a good idea to do this for every single Qt table cell access.

Reply:
Do you have a recommendation on how you would approach this? When I try to use the timer as a member variable, the Dolphin list view doesn't seem to update the times after launching the program even though the ini file is being updated (I'm guessing that the read lock may be interfering here since if I restart Dolphin, the correct times will be shown).

Reply:
Well you need to actually reload the ini at some point so it can see the changed values, and your current class can only do that in the constructor. So make that `ini.Load(ini_path); time_list = ini.GetOrCreateSection("Time Played");` a separate method and then call it when sensible, eg. whenever a game is shut down or paused.

Reply:
```
  connect(&Settings::Instance(), &Settings::EmulationStateChanged, this, [this](Core::State state) {
    if (state == Core::State::Uninitialized) { m_timer.Reload(); }
  });
```

or whatever.